src/Fl_Flex.cxx

```diff
@@ -252,25 +252,31 @@ void Fl_Flex::set_size(Fl_Widget *w, int size) {
     }
   }
 
-  // remove from array ?
+  // remove from array, if we want the widget to be flexible, but an entry was found
   if (size == 0 && idx >= 0) {
-    for (int i = idx; i < set_size_size_ - 2; i++) {
+    for (int i = idx; i < set_size_size_ - 1; i++) {
       set_size_[i] = set_size_[i+1];
     }
     set_size_size_--;
     return;
   }
 
+  // if w is meant to be flexible, we are done now
   if (size == 0)
     return;
 
-  // add to array of fixed size widgets
-  if (set_size_size_ == set_size_alloc_) {
-    set_size_alloc_ = alloc_size(set_size_alloc_);
-    set_size_ = (Fl_Widget **)realloc(set_size_, set_size_alloc_ * sizeof(Fl_Widget *));
+  // if we have no entry yet, add to array of fixed size widgets
+  if (idx==-1) {
+    if (set_size_size_ == set_size_alloc_) {
```

Comment:
Nitpicking, sorry: this should read:
```
if (idx == -1) {
```
for coding style consistency.

Reply:
Fixed in commit 6b650866f2fcf0ffbb64b9f90219aedcd8ec45b1.

Reply:
Yes, hadn't seen that.